lib/storage.js

```diff
@@ -0,0 +1,57 @@
+import _debug from 'debug';
+
+let debug = _debug('secrets:storage');
+
+const NotImplementedError = new Error("Method not implemented");
+
+/**
+ * A base class for deriving storage types.
+ **/
+class BaseStorageObject {
```

Comment:
I wouldn't go down this path... Just require azure credentials when testing locally.
If you have temp TC creds, you can even use http://docs.taskcluster.net/auth/api-docs/#azureTableSAS
which `base.Entity` has support for loading from, when you give it your TC creds... temp creds currently grants access to the `taskclusterdev` storage account... 

If you want to do this, please add a mode to `base.Entity` that support operating with inmemory database instead. Wrapping it just adds additional complexity. IMO using TC creds to get an SAS is the way to go, I know it's sucks to be able to test offline, but you get much better tests. Some errors won't happen unless you use the real APIs.

See `taskcluster-index` for how to use `base.Entity` with TC creds... it does this pretty well...